### PR TITLE
Newsfeed current user

### PR DESCRIPTION
**Pull request by dvidgry**
- used the built in devise method `authenticate_user!` before each post route method.
- used devise `sign_in` test helpers in controller tests.
- stubbed controller tests with a factory girl, factory for User.
- renamed the web helper `sign_in` method to `log_in` because it clashed with the devise built in method for `sign_in`.
- updated feature tests to sign up or sign in before viewing/making posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 class PostsController < ApplicationController
-
   # include FormatUrl
```
Comment (FiddlersCode) [inline app/controllers/posts_controller.rb]: remove comment

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -19,10 +24,12 @@
       post = create(:post)
       expect(Post.find_by(message: "Hello, world!")).to be
```
Comment (FiddlersCode) [inline spec/controllers/posts_controller_spec.rb:25]: Is there an expectation here?

File: spec/features/signed_in_users_see_newsfeed_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+feature 'Newsfeed visible only to signed in users' do
+  scenario 'user signes in and sees newsfeed' do
+    sign_up
+    click_on 'Sign Out'
+    log_in
+    expect(current_path).to eq('/')
+  end
+
+  scenario 'user is not sign in and does not see the newsfeed' do
```
Comment (FiddlersCode) [inline spec/features/signed_in_users_see_newsfeed_spec.rb]: "signed in", not "sign"

**Review by FiddlersCode — CHANGES_REQUESTED** [review]
Nice work!

File: spec/controllers/posts_controller_spec.rb (hunk 2 above)
Comment (anaalta) [reply spec/controllers/posts_controller_spec.rb:25]: it was a default test, you just expect the object to exist. I don't think we need it anymore though

**Comment by sophieklm** [comment]
Looks good, with Paula's changes actioned you may have my :shipit:

**Comment by JessicaBarclay** [comment]
:shipit: from me - one suggestion may be that you announce the change from sign in to log in when merging, incase of problems down the line.

Looks great guys!

**Review by chrisjmit — APPROVED** [review]
:shipit:

**Review by FiddlersCode — APPROVED** [review]

**Comment by FiddlersCode** [comment]
:shipit:

**Pull request merged**